release/ray_release/byod/byod.Dockerfile

```diff
@@ -18,4 +18,4 @@ sudo apt-get update -y \
 EOF
 
 COPY "$PIP_REQUIREMENTS" .
-RUN "$HOME"/anaconda3/bin/pip install --no-cache-dir install -r requirements_byod.txt
+RUN "$HOME"/anaconda3/bin/pip install -r requirements_byod.txt
```

Comment:
why the `--no-cache-dir` needs to be removed?

Reply:
it crashed with this error, i'm not sure why. A little bit search suggests that it could be a pip bug



```
> [5/5] RUN "/home/ray"/anaconda3/bin/pip install --no-cache-dir install -r requirements_byod.txt:
--
  | #14 0.840 Collecting install
  | #14 0.840 ERROR: In --require-hashes mode, all requirements must have their versions pinned with ==. These do not:
  | #14 0.840     install from https://files.pythonhosted.org/packages/4d/c8/8cbca135f9e167810756ea2bc34b028501936675fcbd7dadccf752fa4622/install-1.3.5-py3-none-any.whl
```

Reply:
ah.. I think that is because we are trying to install a compiled `requirements` files in another incompatible environment, probably because the python version is different.

Reply:
and in different python versions, the requirements is different.

which is one of the reasons that using bazel for ray is extra difficult.. because it needs to support multiple python versions at the same time.

Reply:
lol i see

Reply:
so should we do something about this change?

- with `--no-cache-dir` seems to be the right thing to do.
- the ERROR you pointed out seems to be unrelated, but still something we want to resolve.

Reply:
oh, other than --no-cache-dir, the original command has two words install as well, i wonder if that's the source of the issue, i'll check